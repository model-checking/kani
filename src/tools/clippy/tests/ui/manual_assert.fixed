// revisions: edition2018 edition2021
// [edition2018] edition:2018
// [edition2021] edition:2021
// run-rustfix

#![warn(clippy::manual_assert)]
#![allow(clippy::nonminimal_bool)]

fn main() {
    let a = vec![1, 2, 3];
    let c = Some(2);
    if !a.is_empty()
        && a.len() == 3
        && c != None
        && !a.is_empty()
        && a.len() == 3
        && !a.is_empty()
        && a.len() == 3
        && !a.is_empty()
        && a.len() == 3
    {
        panic!("qaqaq{:?}", a);
    }
    assert!(a.is_empty(), "qaqaq{:?}", a);
    assert!(a.is_empty(), "qwqwq");
    if a.len() == 3 {
        println!("qwq");
        println!("qwq");
        println!("qwq");
    }
    if let Some(b) = c {
        panic!("orz {}", b);
    }
    if a.len() == 3 {
        panic!("qaqaq");
    } else {
        println!("qwq");
    }
    let b = vec![1, 2, 3];
    assert!(!b.is_empty(), "panic1");
    assert!(!(b.is_empty() && a.is_empty()), "panic2");
    assert!(!(a.is_empty() && !b.is_empty()), "panic3");
    assert!(!(b.is_empty() || a.is_empty()), "panic4");
    assert!(!(a.is_empty() || !b.is_empty()), "panic5");
}
