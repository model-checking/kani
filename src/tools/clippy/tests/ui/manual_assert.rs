// revisions: edition2018 edition2021
// [edition2018] edition:2018
// [edition2021] edition:2021
// run-rustfix
#![warn(clippy::manual_assert)]

fn main() {
    let a = vec![1, 2, 3];
    let c = Some(2);
    if !a.is_empty()
        && a.len() == 3
        && c != None
        && !a.is_empty()
        && a.len() == 3
        && !a.is_empty()
        && a.len() == 3
        && !a.is_empty()
        && a.len() == 3
    {
        panic!("qaqaq{:?}", a);
    }
    if !a.is_empty() {
        panic!("qaqaq{:?}", a);
    }
    if !a.is_empty() {
        panic!("qwqwq");
    }
    if a.len() == 3 {
        println!("qwq");
        println!("qwq");
        println!("qwq");
    }
    if let Some(b) = c {
        panic!("orz {}", b);
    }
    if a.len() == 3 {
        panic!("qaqaq");
    } else {
        println!("qwq");
    }
    let b = vec![1, 2, 3];
    if b.is_empty() {
        panic!("panic1");
    }
    if b.is_empty() && a.is_empty() {
        panic!("panic2");
    }
    if a.is_empty() && !b.is_empty() {
        panic!("panic3");
    }
    if b.is_empty() || a.is_empty() {
        panic!("panic4");
    }
    if a.is_empty() || !b.is_empty() {
        panic!("panic5");
    }
}
