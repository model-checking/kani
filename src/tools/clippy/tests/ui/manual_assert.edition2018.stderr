error: only a `panic!` in `if`-then statement
  --> $DIR/manual_assert.rs:24:5
   |
LL | /     if !a.is_empty() {
LL | |         panic!("qaqaq{:?}", a);
LL | |     }
   | |_____^ help: try: `assert!(a.is_empty(), "qaqaq{:?}", a);`
   |
   = note: `-D clippy::manual-assert` implied by `-D warnings`

error: only a `panic!` in `if`-then statement
  --> $DIR/manual_assert.rs:27:5
   |
LL | /     if !a.is_empty() {
LL | |         panic!("qwqwq");
LL | |     }
   | |_____^ help: try: `assert!(a.is_empty(), "qwqwq");`

error: only a `panic!` in `if`-then statement
  --> $DIR/manual_assert.rs:44:5
   |
LL | /     if b.is_empty() {
LL | |         panic!("panic1");
LL | |     }
   | |_____^ help: try: `assert!(!b.is_empty(), "panic1");`

error: only a `panic!` in `if`-then statement
  --> $DIR/manual_assert.rs:47:5
   |
LL | /     if b.is_empty() && a.is_empty() {
LL | |         panic!("panic2");
LL | |     }
   | |_____^ help: try: `assert!(!(b.is_empty() && a.is_empty()), "panic2");`

error: only a `panic!` in `if`-then statement
  --> $DIR/manual_assert.rs:50:5
   |
LL | /     if a.is_empty() && !b.is_empty() {
LL | |         panic!("panic3");
LL | |     }
   | |_____^ help: try: `assert!(!(a.is_empty() && !b.is_empty()), "panic3");`

error: only a `panic!` in `if`-then statement
  --> $DIR/manual_assert.rs:53:5
   |
LL | /     if b.is_empty() || a.is_empty() {
LL | |         panic!("panic4");
LL | |     }
   | |_____^ help: try: `assert!(!(b.is_empty() || a.is_empty()), "panic4");`

error: only a `panic!` in `if`-then statement
  --> $DIR/manual_assert.rs:56:5
   |
LL | /     if a.is_empty() || !b.is_empty() {
LL | |         panic!("panic5");
LL | |     }
   | |_____^ help: try: `assert!(!(a.is_empty() || !b.is_empty()), "panic5");`

error: aborting due to 7 previous errors

